src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp

```diff
@@ -143,51 +231,218 @@ struct npc_dashel_stonefistAI : public CombatAI
         {
             SetDeathPrevention(true);
             AttackStart(player);
+        }
+    }
 
-            if (Creature* thug = m_creature->SummonCreature(NPC_OLD_TOWN_THUG, -8672.33f, 442.88f, 99.98f, 3.5f, TEMPSPAWN_DEAD_DESPAWN, 300000))
-                thug->AI()->AttackStart(player);
+    void ResetThug(int thug)
+    {
+        if (thug >= 2)
+            return;
 
-            if (Creature* thug = m_creature->SummonCreature(NPC_OLD_TOWN_THUG, -8691.59f, 441.66f, 99.41f, 6.1f, TEMPSPAWN_DEAD_DESPAWN, 300000))
-                thug->AI()->AttackStart(player);
+        if (Creature* pThug = m_creature->GetMap()->GetCreature(m_thugs[thug]))
+        {
+            if (pThug->IsAlive())
+                pThug->GetMotionMaster()->MovePoint(10, aThugResetPosition[thug][0], aThugResetPosition[thug][1], aThugResetPosition[thug][2], FORCED_MOVEMENT_WALK);
+        }
+    }
+
+    // Dashel returns to his spawn point
+    void JustReachedHome() override
+    {
+        // switch to correct dialog phase, depends if thugs are alive.
+        if (m_dialogStarted)
+        {
+            // This Text gets called no matter thugs are alive
+            DoBroadcastText(SAY_PROGRESS_3_DAS, m_creature);
+
+            if (m_thugsAlive)
+            {
+                // if thugs are alive trigger small rp
+                m_phaseThugEventStage = 1;
+                ResetTimer(DASHEL_END_EVENT, 3000);
+            }
+            else
+            { 
+                // if thugs are dead skip rp and instantly reward quest
+                m_phaseThugEventStage = 5;
+                ResetTimer(DASHEL_END_EVENT, 3000);
+            }                
         }
     }
 
     void HandleEndEvent()
     {
-        SetDeathPrevention(false);
-        DoScriptText(SAY_STONEFIST_3, m_creature);
-        m_creature->CastSpell(nullptr, SPELL_UNKILLABLE_OFF, TRIGGERED_OLD_TRIGGERED);
+        // Occurs only if thugs are alive
+        if(m_thugsAlive)
+        {
+            uint32 timer = 0;
+            switch (m_phaseThugEventStage)
+            {
+                case 1:
+                {
+                    if (Creature* pThug = m_creature->GetMap()->GetCreature(m_thugs[0]))
```

Comment:
name them just thug

Reply:
You mean to change pThug to thug or m_thugs to m_thug?

Reply:
pThug to thug

Reply:
should be done now